subiquity/server/apt.py

```diff
@@ -326,7 +327,8 @@ def _restore_file(path: str) -> None:
         # The file only exists if we are online
         with contextlib.suppress(FileNotFoundError):
             os.unlink(target_mnt.p("etc/apt/sources.list.d/original.list"))
-        _restore_file("etc/apt/sources.list")
+        with contextlib.suppress(FileNotFoundError):
+            _restore_file("etc/apt/sources.list")
```

Comment:
I think we don't want to keep this file on the target system if it wasn't present in the source overlay. I'd suggest something along these lines:

```python
try:
    _restore_file("etc/apt/sources.list")
except FileNotFoundError:
    os.unlink(target_mnt.p("etc/apt/sources.list"))
```